beacon-chain/core/helpers/randao.go

```diff
@@ -80,5 +80,9 @@ func RandaoMix(state *pb.BeaconState, wantedEpoch uint64) ([]byte, error) {
 		return nil, fmt.Errorf("input randaoMix epoch %d out of bounds: %d <= epoch < %d",
 			wantedEpoch, earliestEpoch, currentEpoch)
 	}
-	return state.LatestRandaoMixes[wantedEpoch%params.BeaconConfig().LatestRandaoMixesLength], nil
+	idx := wantedEpoch % params.BeaconConfig().LatestRandaoMixesLength
+	if idx >= uint64(len(state.LatestRandaoMixes)) {
+		return nil, fmt.Errorf("requested id in latest randao mix is out of bound requested id: %v, array length: %v", idx, len(state.LatestRandaoMixes))
```

Comment:
Can you put it as index or idx instead of id

Reply:
changed